packages/botonic-pulumi/src/index.ts

```diff
@@ -2,10 +2,27 @@ import { Config } from '@pulumi/pulumi'
 import { join } from 'path'
 import { cwd } from 'process'
 
-export const getNamePrefix = (): string => {
+export const PROJECT_NAME_SEPARATOR = '-'
+export const MAX_PROJECT_NAME_LENGTH = 30
+
+export function getNamePrefix(): string {
   const config = new Config()
-  return `botonic-${config.get('projectName')}-${config.get('stackName')}`
+  return generatePrefix(
+    config.get('projectName') as string,
+    config.get('stackName') as string
+  )
+}
+
+export function generatePrefix(projectName: string, stackName: string): string {
+  const prefix = `${projectName}${PROJECT_NAME_SEPARATOR}${stackName}`
+  if (prefix.length > MAX_PROJECT_NAME_LENGTH + PROJECT_NAME_SEPARATOR.length) {
+    throw new Error(
+      `The combination of 'projectName' and 'stackName' names can not exceed ${MAX_PROJECT_NAME_LENGTH} chars.`
+    )
```

Comment:
This is the error that will be shown to the developer right? Better be more explicit with something like `` `Provided projectName "${projectName}" and stackName "${stackName}" that combined exceed the max allowed length: ${prefix.length - PROJECT_NAME_SEPARATOR.length} / ${MAX_PROJECT_NAME_LENGTH - PROJECT_NAME_SEPARATOR.length}` ``

Reply:
sounds good, changed 👍